Test mvp

Darkbeast-glitch opened the pull request.

Darkbeast-glitch committed:
Add TeamMember model and related functionality

- Introduced TeamMember model with fields for name, role, short bio, avatar, and social media links.
- Created serializers and views for TeamMember, including a public endpoint to list active team members.
- Updated URL configurations to include team-related routes.
- Added admin configuration for managing TeamMember entries.
- Refactored existing code to ensure proper imports and functionality.

Darkbeast-glitch committed:
Update subproject commit reference in speakwise-frontend-rp

Ezi-code commented on speakwise/speakwise/speakwise/base/team_models.py:
```diff
@@ -0,0 +1,69 @@
+"""Team models for the SpeakWise application."""
```
I think we should create teams app for better code management and just leave base for abstract models.

Ezi-code commented on speakwise/speakwise/speakwise/base/team_models.py:
```diff
@@ -0,0 +1,69 @@
+        help_text="Profile picture of the team member",
+    )
+
+    # Social media links
```
I think we have a social model we could use here, instead of reinventing the wheel.

Ezi-code commented on speakwise/speakwise/speakwise/base/serializers.py:
```diff
@@ -0,0 +1,36 @@
+
+    class Meta:
+        model = TeamMember
+        fields = [
```
Since we are using all the fields here, we could just exclude `created_at` and `updated_at`  to keep the code simple and clean.

Ezi-code commented on speakwise/speakwise/speakwise/base/serializers.py:
```diff
@@ -0,0 +1,36 @@
+        """Get the full URL for the avatar image."""
+        if obj.avatar:
+            request = self.context.get("request")
+            if request:
```
Can we write something like this, to ensure that we don't get an error when request is None?
```suggestion
            request = self.context.get("request") if self.context.get("request") else None
```

Ezi-code reviewed (changes requested):
I think this should rebase on dev after the events refactor PR is merged in. I can see that some changes that were to take place in the events are not in this PR.

Darkbeast-glitch committed:
Enhance authentication permissions and event management

- Updated permission classes to check user roles instead of user types.
- Added `IsAttendeeOrSpeakerOrOrganizerOrAdmin` permission class for broader access.
- Introduced organizer relationship in the Event model.
- Modified event views to restrict event creation and visibility based on user roles.
- Created migrations for the new organizer field and social links model.

Darkbeast-glitch committed:
Enhance file upload functionality by adding event validation and error handling
